config/options.go

```diff
@@ -18,25 +18,34 @@
 // OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
 // THE SOFTWARE.
 
-package jaeger
+package config
 
 import (
-	"testing"
+	"github.com/uber/jaeger-lib/metrics"
 
-	"github.com/stretchr/testify/assert"
+	"github.com/uber/jaeger-client-go"
 )
 
-type expectedMetric struct {
-	metric []string
-	value  int
+// ClientOption is a function that sets some option on the client
+type ClientOption func(c *Configuration)
```

Comment:
nit: it would be cleaner to have this function accept a `*ClientOptions`, and then embed it into `Configuration`